torchvision/csrc/cpu/image/readjpeg_cpu.cpp

```diff
@@ -48,7 +48,10 @@ static void torch_jpeg_init_source(j_decompress_ptr cinfo) {}
 
 static boolean torch_jpeg_fill_input_buffer(j_decompress_ptr cinfo) {
   torch_jpeg_mgr* src = (torch_jpeg_mgr*)cinfo->src;
-  // No more data.  Probably an incomplete image;  just output EOI.
+  // No more data.  Probably an incomplete image;  Raise exception.
+  torch_jpeg_error_ptr myerr = (torch_jpeg_error_ptr)cinfo->err;
+  strcpy(jpegLastErrorMsg, "Image is incomplete or truncated");
+  longjmp(myerr->setjmp_buffer, 1);
```

Comment:
Thanks for fixing this!

For future reference, here is [how TensorFlow implements this function](https://github.com/tensorflow/tensorflow/blob/f742e74da3eaa536dddbe8ebc1cb85ee8de7f312/tensorflow/core/lib/jpeg/jpeg_handle.cc#L117-L142). I'm not sure if there are benefits for one over the other, but it's good for the record.